### ISSUE TITLE
Wrongly converting Rook capture  from UCI

### ISSUE THREAD

**Issue by fabianvdW** [niklasf/shakmaty#34]
```Rust
    let mut position: Chess = Fen::from_str("8/8/8/B2p3Q/2qPp1P1/b7/2P2PkP/4K2R b K - 0 1")
        .unwrap()
        .position(CastlingMode::Standard)
        .unwrap();
    Uci::from_str("g2h1")
        .ok()
        .and_then(|x| x.to_move(&position).ok())
        .unwrap();
```
This crashes:  ```called `Option::unwrap()` on a `None` value'```. My initial guess is that is has to do something with capturing the Rook which is relevant for castling, since removing the castling flag no longer leads to crashes.

**Comment by fabianvdW** [comment]
Caused by this line https://github.com/niklasf/shakmaty/blob/822fe9defbbfddd77130f0f9c1ed469c269b9162/src/uci.rs#L308, condition is obviously wrong for the enemy King

**Comment by niklasf** [comment]
Thanks for the clear report. I'll review the code for similar issues and then do a bugfix release.

**Comment by niklasf** [comment]
Published in v0.17.2.
